Automattic/simplenote-android
# Update Navigation Drawer

### Fix
Update the navigation drawer interface from a custom implementation to the standard `NavigationView` for better compatibility with all Android variants.  The ***Settings*** item was also moved from the bottom of the navigation drawer to the top group with ***All Notes*** and ***Trash*** for better organization.  When no custom tags exist, the ***Tags*** group and ***Untagged Notes*** item are hidden.  See the screenshots below for illustration.

![update_navigation_drawer_tags_light](https://user-images.githubusercontent.com/3827611/65538961-b9e18900-dec5-11e9-9924-c9563d97b0c3.png)

![update_navigation_drawer_tags_dark](https://user-images.githubusercontent.com/3827611/65538968-bbab4c80-dec5-11e9-965d-1502d1f83a95.png)

![update_navigation_drawer_tagless_light](https://user-images.githubusercontent.com/3827611/65538974-be0da680-dec5-11e9-877a-e6ff981b92b9.png)

![update_navigation_drawer_tagless_dark](https://user-images.githubusercontent.com/3827611/65538976-c0700080-dec5-11e9-9f57-6a848a5dcb0a.png)

### Test
1. Use account without custom tags.
2. Open navigation drawer.
3. Notice ***All Notes***, ***Trash***, and ***Settings*** items shown in navigation drawer.
4. Notice ***Tags*** group and ***Untagged Notes*** item hidden in navigation drawer.
5. Close navigation drawer.
6. Tap any note in list.
7. Add custom tag in ***Add tag...*** field.
8. Tap back arrow in app bar.
9. Open navigation drawer.
10. Notice ***Tags*** group and ***Untagged Notes*** item shown in navigation drawer.
11. Tap custom tag item in navigation drawer.
12. Notice note(s) with custom tag shown in list.
13. Open navigation drawer.
14. Tap ***Untagged Notes*** item in navigation drawer.
15. Notice untagged notes shown in list.
16. Open navigation drawer.
17. Tap ***Edit*** button in ***Tags*** group.
18. Notice custom tag is shown in list on ***Edit tags*** screen.
19. Tap the ***Delete*** action next to custom tag in list.
20. Tap ***Yes*** button in ***Delete tag*** dialog.
21. Notice custom tag is removed from list on ***Edit tags*** screen.
22. Tap back arrow in app bar.
23. Open navigation drawer.
24. Notice ***Tags*** group and ***Untagged Notes*** item hidden in navigation drawer.
25. Tap ***Trash*** item in navigation drawer.
26. Notice trashed notes shown in list.
27. Tap back arrow in app bar.
28. Open navigation drawer.
29. Tap ***Settings*** item in navigation drawer.
30. Notice ***Settings*** screen is shown.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in c0cb6eae with:
> Updated navigation drawer interface for better organization and compatibility
